packages/autoloader/src/functions.php

```diff
@@ -134,38 +134,111 @@ function autoloader( $class_name ) {
+}
+
+/**
+ * Returns an array containing the active plugins. If plugin is activating, it
+ * is included in the array.
+ *
+ * @return Array An array of plugin names as strings.
+ */
+function get_active_plugins() {
+	$active_plugins = (array) get_option( 'active_plugins', array() );
+	$current_plugin = get_current_plugin();
 
-	$include_files = file_exists( $autoload_file ) ? require $autoload_file : array();
+	if ( ! in_array( $current_plugin, $active_plugins, true ) ) {
+		// The current plugin isn't active, so it must be activating. Add it to the list.
+		$active_plugins[] = $current_plugin;
+	}
 
-	foreach ( $include_files as $file_identifier => $file_data ) {
-		enqueue_package_file( $file_identifier, $file_data['version'], $file_data['path'] );
+	// If the activating plugin is not the current plugin, add it to the list.
+	$activating_plugin = get_activating_plugin();
+	if ( $activating_plugin && $current_plugin !== $activating_plugin ) {
+		$active_plugins[] = $activating_plugin;
 	}
 
-	// TODO: The plugins_loaded action checks aren't necessary anymore.
-	if (
-		function_exists( 'has_action' )
-		&& function_exists( 'did_action' )
-		&& ! did_action( 'plugins_loaded' )
-		&& false === has_action( 'plugins_loaded', __NAMESPACE__ . '\file_loader' )
-	) {
-		// Add action if it has not been added and has not happened yet.
-		// Priority -10 to load files as early as possible in case plugins try to use them during `plugins_loaded`.
-		add_action( 'plugins_loaded', __NAMESPACE__ . '\file_loader', 0, -10 );
+	return $active_plugins;
+}
+
+/**
+ * Creates an array containing the paths to the classmap and filemap for the given plugin.
+ * The filenames are the names of the files generated by the Jetpack Autoloader version >2.0.
+ *
+ * @param String $plugin The plugin string.
+ * @return Array An array containing the paths to the plugin's classmap and filemap.
+ */
+function create_map_path_array( $plugin ) {
+	$plugin_path = plugin_dir_path( trailingslashit( WP_PLUGIN_DIR ) . $plugin );
+
+	return array(
+		'class' => trailingslashit( $plugin_path ) . 'vendor/composer/jetpack_autoload_classmap.php',
+		'file'  => trailingslashit( $plugin_path ) . 'vendor/composer/jetpack_autoload_filemap.php',
+	);
+}
+
+/**
+ * Checks whether the current plugin is active.
+ *
+ * @return Boolean True if the current plugin is active, else false.
+ */
+function is_current_plugin_active() {
+	$active_plugins = (array) get_option( 'active_plugins', array() );
+	$current_plugin = get_current_plugin();
+
+	return in_array( $current_plugin, $active_plugins, true );
+}
+
+/**
+ * Returns the name of activating plugin if a plugin is activating via a request.
+ *
+ * @return String The name of the activating plugin.
+ */
+function get_activating_plugin() {
+	if ( isset( $_REQUEST['action'] ) && 'activate' === $_REQUEST['action'] ) {
+		$activating_plugin = isset( $_REQUEST['plugin'] ) ? wp_unslash( $_REQUEST['plugin'] ) : null;
+		return $activating_plugin;
+	}
```

Comment:
This probably needs a nonce verification.

Reply:
You're right, but it turns out that the function that checks the validity of the nonce is not yet loaded at that time. So I have added a "check" that only requires the nonce to be present in the request, but doesn't actually check it. The request will fail later when the plugin is activated.

Reply:
This looks good to me!